Docker/Dockerfile

```diff
@@ -0,0 +1,132 @@
+# Copyright (c) Contributors to the Open 3D Engine Project.
+# For complete copyright and license terms please see the LICENSE at the root of this distribution.
+#
+# SPDX-License-Identifier: Apache-2.0 OR MIT
+#
+
+# The cpu architecture 
+ARG INPUT_ARCHITECTURE=amd64
+
+# The root docker image 
+ARG INPUT_IMAGE=ubuntu
+
+# The root docker tag 
+ARG INPUT_TAG=jammy
+
+# Pull in the base ubuntu docker
+FROM ${INPUT_ARCHITECTURE}/${INPUT_IMAGE}:${INPUT_TAG}
+
+ARG INPUT_IMAGE
+ARG INPUT_TAG
+
+ENV INPUT_IMAGE=$INPUT_IMAGE
+ENV WORKSPACE=/data/workspace
+
+WORKDIR $WORKSPACE
+
+# O3DE Arguments
+ARG O3DE_REPO=https://github.com/o3de/o3de
+ARG O3DE_BRANCH=development
+ARG O3DE_COMMIT=HEAD
+
+# O3DE Environment
+ENV O3DE_REPO=$O3DE_REPO
+ENV O3DE_BRANCH=$O3DE_BRANCH
+ENV O3DE_COMMIT=$O3DE_COMMIT
+ENV O3DE_ROOT=$WORKSPACE/o3de
+
+ENV LANG=en_US.UTF-8
+
+# Validate the tag if the image is 'ubuntu'
+RUN if [ "$INPUT_IMAGE" = "ubuntu" ]; then \
+        if [ "$INPUT_TAG" != "jammy" ] && [ "$INPUT_TAG" != "nobel" ]; then \
+            echo "ERROR: Unsupported ubuntu tag: ${INPUT_TAG}. Must be either 'jammy' or 'nobel'"; \
+            exit 1;\
+        fi \
+    fi
+
+# Validate the tag if the image is 'ros'
+RUN if [ "$INPUT_IMAGE" = "ros" ]; then \
+        if [ "$INPUT_TAG" != "humble" ] && [ "$INPUT_TAG" != "jazzy" ]; then \
+            echo "ERROR: Unsupported ros tag: ${INPUT_TAG}. Must be either 'humble' or 'jazzy'"; \
+            exit 1;\
+        fi \
+    fi
+
+# Setup time zone and locale data (necessary for SSL and HTTPS packages)
+RUN apt-get update \
+    && DEBIAN_FRONTEND="noninteractive" apt-get -y install tzdata locales keyboard-configuration curl \
+    && sed -i -e 's/# en_US.UTF-8 UTF-8/en_US.UTF-8 UTF-8/' /etc/locale.gen \
+    && dpkg-reconfigure --frontend=noninteractive locales \
+    && update-locale LANG=en_US.UTF-8 \
+    && rm -rf /var/lib/apt/lists/*
+
+# Install the required ubuntu packages
```

Comment:
I see that you check for image OS type (i.e. ubuntu and ros), but are there differences between apt-dependencies for each OS?

Reply:
The `ros` docker images are [derivatives](https://hub.docker.com/_/ros) of ubuntu.